### ISSUE TITLE
[BUG] replaceViewController problems with modalPresentationStyles

### ISSUE THREAD

**Issue by FSurani** [HeroTransitions/Hero#532]
#198 
So I have a bit of a problem. I have two ViewControllers that need to be presented one after another and I'd like to chain them. But I can't quite get it to work. My code is as follows:
```
let src = FirstViewController()
src.hero.isEnabled = true
src.modalPresentationStyle = .overCurrentContext //IMPORTANT BIT
tabbar.present(src, animated: true, completion: nil)

... //some place else ...

let dst = SecondViewController()
let dstNav = UINavigationController.init(rootViewController: dst)
dst.hero.isEnabled = true

src.hero.replaceViewController(with: dstNav)
```

Now. The above works great for the transition. But when I do `self.dismiss` from dst it doesn't do anything. 

EDIT: I also noticed that if I add a completion handler for the replaceViewController method it also never fires.

**Comment by FSurani** [comment]
Apparently there has been another issue with this: #509. I'm also seeing the same.

**Comment by FSurani** [comment]
Okay so I finally found out WHAT was broken!
So my issue was that my `src` had it's modalAnimationType set to overFullScreen. When I changed it to overCurrentContext it fixed it! Including #509 . 

This ticket helped #502 

I'm leaving this open in case for now.

**Comment by FSurani** [comment]
Created a repo that has the issue and an example of it.

https://github.com/FSurani/HeroContextProblem
